inlet/metadata/provider/root.go

```diff
@@ -32,8 +36,18 @@ type BatchQuery struct {
 
 // Answer is the answer received from a provider.
 type Answer struct {
-	ExporterName string
-	Interface
+	ExporterName          string
+	ExporterRegion        string
+	ExporterRole          string
+	ExporterTenant        string
+	ExporterSite          string
+	ExporterGroup         string
+	InterfaceName         string
+	InterfaceDescription  string
+	InterfaceSpeed        uint
+	InterfaceProvider     string
+	InterfaceConnectivity string
+	InterfaceBoundary     schema.InterfaceBoundary
```

Comment:
Having substructures was cleaner, wasn't it? Notably if you didn't do that because of the `string`/`InterfaceBoundary` mismatch.

Reply:
That was the main reason yes :) 

Also I figured this was one less struct copied around (dubious argument I agree !).

In the end, I also think that it's complicated to share structs between user input and internal code needs.
For example, there is `ExporterConfiguration` for internal needs (with ifindexes lookup),
but it cannot be loaded from json because json does not support `map[uint]`, only `map[string]`, so there is `exporterInfo` with an `[]exporterInterfaces` (which is basically the same as Interface, but with the ifindex pushed down). In the same way, `exporterInfo` is also required because jq requires that we end up with a list of exporters, and not a map (`ExporterSubnet` is pushed down compared to loading from the config).
But at the same time, this leads to somewhat duplicated code...

Reply:
I agree with your last paragraph: let's not try that if it introduces more complexity. Let me think about the other comment.

Reply:
Note that you can definitely load anything from JSON by implementing `UnmarshalJSON()` method. It really depends on what solution is the simplest. We can postpone this question to later too, as long as the test coverage is OK, I should be able to change it if it bothers me too much.